modules/udf/main.tf

```diff
@@ -14,7 +14,16 @@
  * limitations under the License.
  */
 
-module "bq_find_in_set" {
+ locals {
+   // Regex to extract function name (optionally might include project and dataset ids)
+   // see https://cloud.google.com/bigquery/docs/reference/standard-sql/user-defined-functions#temporary-udf-syntax
+   // You can view test cases at: https://regex101.com/r/a7CNA6/2/
```

Comment:
Let's make this is tested in our tests somehow (I know it's some extra work, but regex is notoriously easy to break + hard to get right).

Probably the easiest way would be to add those test cases you provided as an integration test (in `test/fixtures`) and check the value of `added_udfs`.

Reply:
Sound right to me. 
I've added them to the multi table example. 
I'm not familiar with ruby / this "controls" syntax or inspec.
Is there an easy way to just check the value of module output? or good getting started materials?

Reply:
@jaketf so it involves a couple of steps, [here](https://github.com/terraform-google-modules/terraform-google-cloud-router/pull/11) is an example PR.
i) plumb the output `bigquery_udfs` from the example to the [fixture output](https://github.com/terraform-google-modules/terraform-google-bigquery/blob/master/test/fixtures/full/outputs.tf)
ii) update [inspec.yml](https://github.com/terraform-google-modules/terraform-google-bigquery/blob/master/test/integration/full/inspec.yml) with the plumbed output name from the fixture
iii) You can retrieve the output in rb like [this](https://github.com/terraform-google-modules/terraform-google-bigquery/blob/392f735c05f7305743792017eff0f792d62ea5dd/test/integration/full/controls/big_query.rb#L18) 

Regarding the test itself, you could probably just hard code and assert expected map, perhaps something like [this](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/882552001f3c4a1206d23b5c54a2e9110983173e/test/integration/safer_cluster/controls/gcloud.rb#L50).

Reply:
Btw we have a codelab for tests if you haven't seen it before: https://codelabs.developers.google.com/codelabs/cft-onboarding/#6

Reply:
So interestingly, this behavior is a bit tough to integration test because we need "another project" and "another dataset" that fixed across all tests in order to infer project / dataset from the from DDL OR do something confusing in the example.

I think this is actually good signal that we should force users to have only function name (no project / dataset id prefix) in their DDL. This should just be explicitly handled by the module project_id / dataset_id. Otherwise the tests and usage might quickly get confusing.

Reply:
I'm not entirely sure I follow the need for another project/dataset to be referenced? It looks like the way you have it should work.